src/decide.ts

```diff
@@ -63,6 +129,31 @@ export class Decide {
             return
         }
 
-        this.instance._afterDecideResponse(response)
+        this.instance._onRemoteConfig(response)
+    }
+
+    private onRemoteConfig(config?: RemoteConfig): void {
+        // NOTE: Once this is rolled out we will remove the "decide" related code above. Until then the code duplication is fine.
+        if (!config) {
+            logger.error('Failed to fetch remote config from PostHog.')
+            return
+        }
+        if (!(document && document.body)) {
+            logger.info('document not ready yet, trying again in 500 milliseconds...')
+            setTimeout(() => {
+                this.onRemoteConfig(config)
+            }, 500)
+            return
+        }
+
+        this.instance._onRemoteConfig(config)
+
+        if (config.hasFeatureFlags !== false) {
+            // TRICKY: This is set in the parent for some reason...
+            this.instance.featureFlags.setReloadingPaused(false)
+            // If the config has feature flags, we need to call decide to get the feature flags
+            // This completely separates it from the config logic which is good in terms of separation of concerns
+            this.instance.featureFlags.reloadFeatureFlags()
```

Comment:
don't really need to change this now... i wonder how much order of operations is important here.

would be nice for `this.instance.featureFlags` to be listening to `onRemoteConfig` and run these things itself to break this dependency

but only noticing it because this has all moved

Reply:
Yeah I'd be tempted to clean this up later. The main thing is I really dont want to do fundamental changes whilst we still rely on the old decide path.